### ISSUE TITLE
New faculty status handling in the menu

### ISSUE THREAD

**Issue by mwvolo** [openstax/os-webview#2530]
We have added a few new faculty status options to the accounts user API. The full list is:
```
no_faculty_info
incomplete_signup
pending_sheerid
pending_faculty
confirmed_faculty
rejected_by_sheerid
rejected_faculty
```

Previously, we had shown menu options ("Confirm Your Email" or "Request Instructor Access") based on this status.

- [ ] "Request Instructor Access" should be shown to everyone not `confirmed_faculty` or `rejected_faculty`
- [ ] A new option, "Complete Your Profile" should be shown to `incomplete_signup`. Directing them to accounts will put them in the correct spot for them to finish signing up.
- [ ] "Confirm Your Email" should be shown if they don't have an email in `contact_infos` that is confirmed.

Example of `contact_infos` that would qualify to see the "Confirm Your Email" option:
```
"contact_infos": [
    {
      "id": 123,
      "type": "EmailAddress",
      "value": "test@rice.edu",
      "is_verified": false,
      "is_guessed_preferred": true
    }
  ],
```

**Comment by RoyEJohnson** [comment]
@mwvolo 
For clarity: "Request instructor access" _and_ "Complete your profile" will both show if status is `incomplete_signup`?
And "Verify your email address" can also show at the same time?
All three are active in the situation I'm testing:

![Image](https://github.com/openstax/os-webview/assets/7315241/bd49d247-f5db-4620-a364-0fbaa435f142)

**Comment by mwvolo** [comment]
No, we should only show "Request instructor access" when a user has `no_faculty_info`. Sorry, forgot that caveat.

For "Verify your email address", are you checking if they have a confirmed email address? That should rarely be shown to anyone. If you're seeing it with an account on dev, can I get that account id?

**Comment by mwvolo** [comment]
If they have a `pending_faculty` status, showing them they are pending (_Pending Verification_, non-clickable) might be nice.
I feel like we used to do something like this, but could be misremembering.

**Comment by RoyEJohnson** [comment]
We have a "Pending instructor access" link (goes to profile) for when the `faculty_status` is `rejected_by_sheerid`.
I can change that to `pending_faculty` or have it for both cases.

**Comment by mwvolo** [comment]
Perfect, yes, that can be for both of those statuses.
